src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore/Infrastructure/GlobalExceptionHandler.cs

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Net;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using EdFi.DmsConfigurationService.Backend;
+using Microsoft.AspNetCore.Diagnostics;
+
+namespace EdFi.DmsConfigurationService.Frontend.AspNetCore.Infrastructure;
+
+public sealed class GlobalExceptionHandler(ILogger<GlobalExceptionHandler> logger) : IExceptionHandler
+{
+    public async ValueTask<bool> TryHandleAsync(
+        HttpContext httpContext,
+        Exception exception,
+        CancellationToken cancellationToken
+    )
+    {
+        var traceId = httpContext.TraceIdentifier;
+        logger.LogError(
```

Comment:
Question for the team, do we really want an ERROR log and full stack trace for all exceptions? For example, Validation exceptions will be logging the following for simple validation errors:

```
2024-11-06 12:39:30.274 ERR An unhandled exception has occurred while executing the request.
FluentValidation.ValidationException: Validation failed: 
 -- ContactName: 'Contact Name' must not be empty. Severity: Error
 -- NamespacePrefixes: 'Namespace Prefixes' must not be empty. Severity: Error
   at EdFi.DmsConfigurationService.Frontend.AspNetCore.Infrastructure.ValidatorExtensions.GuardAsync[TRequest](IValidator`1 validator, TRequest request) in C:\Tanager\Data-Management-Service\src\config\frontend\EdFi.DmsConfigurationService.Frontend.AspNetCore\Infrastructure\ValidatorExtensions.cs:line 18
   at EdFi.DmsConfigurationService.Frontend.AspNetCore.Modules.VendorModule.InsertVendor(VendorInsertCommandValidator validator, VendorInsertCommand entity, HttpContext httpContext, IVendorRepository repository) in C:\Tanager\Data-Management-Service\src\config\frontend\EdFi.DmsConfigurationService.Frontend.AspNetCore\Modules\VendorModule.cs:line 37
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.ExecuteTaskResult[T](Task`1 task, HttpContext httpContext)
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.<>c__DisplayClass102_2.<<HandleRequestBodyAndCompileRequestDelegateForJson>b__2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|7_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddlewareImpl.<Invoke>g__Awaited|10_0(ExceptionHandlerMiddlewareImpl middleware, HttpContext context, Task task)
```